src/CredentialsImport.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Use this as place to store any state or functionality related to password import promotion
+ */
+class CredentialsImport {
+    /** @param {import("./DeviceInterface/InterfacePrototype").default} device */
+    constructor (device) {
+        this.device = device
+    }
+
+    /**
+     * Check if password promotion prompt should be shown
+     */
+    shouldShowCredentialsImportFlow () {
+        const {credentials, credentialsImport} = this.device.settings.availableInputTypes
+        if (credentials?.username && credentials?.password) return false
+        return credentialsImport
+    }
+
+    async init () {
+        // Attach callbacks and listneres for response from the native side.
+    }
+
+    async started () {
+        this.device.firePixel({pixelName: 'autofill_import_credentials_flow_started'})
```

Comment:
Isn't this a native-side pixel? I think we should leave the firing logic entirely to them in this case. We're already sending a message to open the import flow, so they can fire it from there.

Reply:
Yep removed https://github.com/duckduckgo/duckduckgo-autofill/pull/631/commits/03366ad416641e15bf246cddafef8d9b5cbee473